exercises/concept/secure-treasure-chest/.docs/instructions.md

```diff
@@ -0,0 +1,56 @@
+# Instructions
+
+In this exercise you're going to write a generic (/ magical!) SecureTreasureChest, to store some treasure.
+Sharp eyed students will have noticed that the previous `TreasureChest` wasn't that secure!
+If you used the `getTreasure` function, then you had to supply the password, but you could still simply destructure the `TreasureChest` type to get the treasure, without having to know the password.
+Even worse, you could even retrieve the password in the same way!
+Let's fix that by using an Opaque Type.
+
+## 1. Define the Password opaque type, using the "Parse, don't validate" pattern
+
+Define a `Password` custom type with a single variant.
+
+The variant should have an associated `String` value, for the password.
+
+You should expose the `Password` type, but not the variant. This makes `Password` an Opaque type, and means that you can no longer use destructing to get the password.
+
+Define and expose a `createPassword` function that takes a `String` and returns a `Maybe Password`.
```

Comment:
Should we ask them to add a signature for this function (and the others?).
We can check that with the analyzer.